http3/client.go

```diff
@@ -114,14 +119,16 @@ func newClient(hostname string, tlsConf *tls.Config, opts *roundTripperOpts, con
 	tlsConf.NextProtos = []string{versionToALPN(conf.Versions[0])}
 
 	return &client{
-		hostname:      authorityAddr("https", hostname),
-		tlsConf:       tlsConf,
-		requestWriter: newRequestWriter(logger),
-		decoder:       qpack.NewDecoder(func(hf qpack.HeaderField) {}),
-		config:        conf,
-		opts:          opts,
-		dialer:        dialer,
-		logger:        logger,
+		hostname:         authorityAddr("https", hostname),
+		tlsConf:          tlsConf,
+		requestWriter:    newRequestWriter(logger),
+		decoder:          qpack.NewDecoder(func(hf qpack.HeaderField) {}),
+		config:           conf,
+		opts:             opts,
+		dialer:           dialer,
+		receivedGoawayID: quic.StreamID(-4),
```

Comment:
This seems hacky. Introduce a protocol.InvalidStreamID instead?

Reply:
I got the stream ID from [rfc 9000](https://www.rfc-editor.org/rfc/rfc9000.html#name-stream-types-and-identifier). For different kinds of streams, the criteria for invalid id is different. I think a function that verifies ID would be better instead of a new type.